viadot/sources/vid_club.py

```diff
@@ -0,0 +1,165 @@
+    There are 4 endpoints where to get the data.
+
+    """
+
+    def __init__(self, *args, credentials: Dict[str, Any] = None, **kwargs):
+        """
+        Create an instance of VidClub.
+
+        Args:
+            credentials (dict, optional): Credentials to Vid Club APIs.
+                Defaults to dictionary.
+        """
+
+        DEFAULT_CREDENTIALS = local_config.get("VIDCLUB")
+        credentials = kwargs.pop("credentials", DEFAULT_CREDENTIALS)
+        if credentials is None:
+            raise CredentialError("Missing credentials.")
+
+        self.headers = {
+            "Authorization": "Bearer " + credentials["token"],
+            "Content-Type": "application/json",
+        }
+
+        super().__init__(*args, credentials=credentials, **kwargs)
+
+    def build_query(
+        self,
+        source: Literal["jobs", "product", "company", "survey"],
+        from_date: str,
+        to_date: str,
+        api_url: str,
+        items_per_page: int,
+    ) -> str:
+        """
+        Builds the query from the inputs.
+
+        Args:
+            source (str): The endpoint source to be accessed, has to be among these:
+                ['jobs', 'product', 'company', 'survey'].
+            from_date (str): Start date for the query.
+            to_date (str): End date for the query, if empty, datetime.today() will be used.
+            api_url (str): Generic part of the URL.
+            items_per_page (int): number of entries per page.
+
+        Returns:
+            str: Final query with all filters added.
+        """
+        if source in ["jobs", "product", "company"]:
+            url = f"{api_url}{source}?from={from_date}&to={to_date}&limit={items_per_page}"
+        elif source in "survey":
+            url = f"{api_url}{source}?language=en&type=question"
+        else:
+            raise ValidationError(
+                "Pick one these sources: jobs, product, company, survey"
+            )
+        return url
+
+    def get_response(
+        self,
+        source: Literal["jobs", "product", "company", "survey"],
+        from_date: str = "2022-03-22",
+        to_date: str = None,
+        items_per_page: int = 100,
+        region="null",
+    ) -> pd.DataFrame:
+        """
+        Gets the response from the API queried and transforms it into DataFrame.
+
+        Args:
+            source (str): The endpoint source to be accessed, has to be among these:
+                ['jobs', 'product', 'company', 'survey'].
+            from_date (str, optional): Start date for the query, by default is the oldest date in the data 2022-03-22.
+            to_date (str, optional): End date for the query. By default, datetime.today() will be used.
+            items_per_page (int, optional): Number of entries per page. 100 entries by default.
+            region (str, optinal): Region filter for the query. By default, it is empty.
+
+        Returns:
+            pd.DataFrame: Table of the data carried in the response.
```

Comment:
Please add `Raises` section in the docstring. In this function, you have `ValidationError`.